mantidproject/mantid
# ISIS Powder Mask out prompt pulses for HRPD

When loading data using HRPD's long frame, the prompt pulse is now masked out. This was causing problems for runs where the chopper malfunctioned, and somehow got missed off porting over HRPD from Aziz's scripts.

**To test:**
If not reviewing from ISIS, you'll need to change the second line and do some setup locally - ping me message for an explanation of how to set up ISIS_Powder directories, as it can be fiddly.
```
from isis_powder import HRPD

hrpd = HRPD(config_file=r"\\olympic\babylon5\Public\JoeRamsay\hrpd\hrpd_config.yaml")

hrpd.create_vanadium(window="100-200", first_cycle_run_no=68737)
hrpd.focus(run_number=69201, window="100-200")
```

Run this on master first - `69201.raw-ResultTOF-1` should look strange, with a section with negative counts:
![capture](https://user-images.githubusercontent.com/29916298/36030728-37ca861c-0da0-11e8-853b-bc0d09748888.PNG)

Checkout the branch and rerun the script (you'll have to restart Mantid). The workspace should now look more sane:
![capture](https://user-images.githubusercontent.com/29916298/36030807-84e1172c-0da0-11e8-85dc-d0dc42d46d7f.PNG)

Fixes #21722 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/commit/264f080f7db94ee7450f956a3472d5b55c0809d9)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
